### DISCUSSION TITLE
Production environment variables not reflected

### DISCUSSION THREAD

**Discussion by jeepers3327** [vercel/next.js#25189] [Help]
I deployed my nextjs app through vercel and configure environment variables to be used for API and WS endpoints.

![image](https://user-images.githubusercontent.com/70475641/118491865-fffdf700-b751-11eb-8240-6bc764a9af4c.png) (My apologies for redundant env variables, will remove duplicate env after solving this issue)

When I visit the production and deployed site, it still uses `localhost` urls.
![image](https://user-images.githubusercontent.com/70475641/118492208-6125ca80-b752-11eb-9b93-77185ca1f75e.png)

I tried `redeploy` or `promote to production` but still it doesn't reflect.
Do I have to do something in order for these variables to reflect?

Thank you.

**Comment by wwwdepot** [comment]
Did you hardcode any url string? Please provide a minimum reproduction.

**Reply by jeepers3327** [reply]
Here's my [repo](https://github.com/jeepers3327/senti). Nextjs project is on `app` directory.

**Reply by wwwdepot** [reply]
Prefix environment variables with `NEXT_PUBLIC_` if you're using them in browsers. https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

```jsx
// _app.tsx
const APP_WS_URL = process.env.NEXT_PUBLIC_APP_WS_URL || `ws://localhost:4000/socket`;
```

**Reply by jeepers3327** [reply]
Okay I understand,  as you can see it in the screenshot, on the error which says CORS error, it uses localhost:3000 instead of the url set on vercel.

**Comment by rlenoir-codepoint** [comment]
Hello @jeepers3327 , did you solve your issue since you created this discussion? We are facing a similar problem in our current project, where `production` as well as `preview - staging` is picking up the `preview - dev` values.

**Reply by jeepers3327** [reply]
No, unfortunately. I tried recreating the project in vercel and that seems fix the issue. 

Maybe you can create your own discussion with details and hopefully someone from Nextjs team can answer it.

**Reply by rlenoir-codepoint** [reply]
Thanks for your prompt reply, I am currently exchanging with the Vercel support. I will post any useful information here once the problem is solved.
